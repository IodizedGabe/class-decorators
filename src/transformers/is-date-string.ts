import { composePropertyDecorators, IsDateStringOptions } from '@hippo-oss/dto-decorators';

import { createBasePropertyDecorators } from './base';

export function IsDateString(options: IsDateStringOptions = {}): PropertyDecorator {
    return composePropertyDecorators([
        ...createBasePropertyDecorators(options),
    ]);
}
