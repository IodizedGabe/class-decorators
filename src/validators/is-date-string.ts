import { composePropertyDecorators, IsDateStringOptions } from '@hippo-oss/dto-decorators';
import { IsISO8601 } from 'class-validator';

import { createBasePropertyDecorators } from './base';

export function IsDateString(options: IsDateStringOptions): PropertyDecorator {
    return composePropertyDecorators([
        ...createBasePropertyDecorators(options),

        // validate data as a DateString
        IsISO8601({}, { each: !!options.isArray }),
    ]);
}
