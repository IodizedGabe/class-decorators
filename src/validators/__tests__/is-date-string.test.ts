import { validateSync } from 'class-validator';

import { IsDateString } from '../is-date-string';

describe('IsDateString', () => {
    it('allows optional date string to be omitted', () => {
        class Fixture {
            @IsDateString({
                optional: true,
            })
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
    it('allows nullable date string to be omitted', () => {
        class Fixture {
            @IsDateString({
                nullable: true,
            })
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
    it('does not allow required date string to be omitted', () => {
        class Fixture {
            @IsDateString({})
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(1);
        expect(errors).toMatchSnapshot();
    });
    it('does not allow non-date string', () => {
        class Fixture {
            @IsDateString({})
            property!: string;
        }

        const obj = new Fixture();
        obj.property = 'foo';

        const errors = validateSync(obj);
        expect(errors).toHaveLength(1);
        expect(errors).toMatchSnapshot();
    });
    it('allows date string', () => {
        class Fixture {
            @IsDateString({})
            property!: string;
        }

        const obj = new Fixture();
        obj.property = '2022-03-10';
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
    it('allows date time string', () => {
        class Fixture {
            @IsDateString({})
            property!: string;
        }

        const obj = new Fixture();
        obj.property = '2022-03-11T00:37:19.486Z';
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
});
