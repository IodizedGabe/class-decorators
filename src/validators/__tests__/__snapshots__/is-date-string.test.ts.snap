// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsDateString does not allow non-date string 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isIso8601": "property must be a valid ISO 8601 date string",
    },
    "property": "property",
    "target": Fixture {
      "property": "foo",
    },
    "value": "foo",
  },
]
`;

exports[`IsDateString does not allow required date string to be omitted 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "property should not be null or undefined",
      "isIso8601": "property must be a valid ISO 8601 date string",
    },
    "property": "property",
    "target": Fixture {},
    "value": undefined,
  },
]
`;
